frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2864,6 +2864,45 @@ resolveFunctionByPoisQuery(ResolutionContext* rc, PoiInfo::Trace poiTrace) {
   return CHPL_RESOLUTION_QUERY_END(ret);
 }
 
+// Why does this query exist in addition to resolveFunctionByInfoQuery?
+//
+// Overall, the goal is to enable generic instantiation caching. There are,
+// in fact, two types of PoiInfo: resolved and unresolved ones, which
+// correspond loosely to "before instantiation" and "after instantiation".
```

Comment:
Maybe say "there two kinds of `PoiInfo`" trying to emphasize that they are logically different but inhabit the same type.